_inc/client/pro-status/index.jsx

```diff
@@ -123,141 +127,65 @@ const ProStatus = React.createClass( {
 			if ( 'backups' === feature ) {
 				if ( hasFree && ! hasBackups ) {
 					if ( this.props.isCompact ) {
-						return this.getScanActions( 'free' );
+						return this.getProActions( 'free' );
 					}
 				}
 			}
 
 			if ( 'scan' === feature ) {
 				if ( ( hasFree || hasPersonal ) && ! hasScan ) {
 					if ( this.props.isCompact ) {
-						return this.getScanActions( 'free' );
+						return this.getProActions( 'free' );
 					}
 					return '';
 				}
 				if ( 'N/A' !== vpData ) {
 					const threatsCount = this.props.getScanThreats();
 					if ( 0 !== threatsCount ) {
-						return this.getScanActions( 'threats' );
+						return this.getProActions( 'threats' );
 					}
 					if ( 0 === threatsCount ) {
-						return this.getScanActions( 'secure' );
+						return this.getProActions( 'secure' );
 					}
 				}
 			}
 
 			if ( 'akismet' === feature ) {
-				const akismetData = this.props.getAkismetData();
-				if ( 'invalid_key' === akismetData ) {
-					return (
-						<a href={ this.props.siteAdminUrl + 'admin.php?page=akismet-key-config' } >
-							<SimpleNotice
-								showDismiss={ false }
-								status='is-warning'
-								isCompact={ true }
-							>
-								{ __( 'Invalid key', { context: 'Short warning message about an invalid key being used for Akismet.' } ) }
-							</SimpleNotice>
-						</a>
-					);
-				}
-			}
-
-			if ( 'seo-tools' === feature ) {
-				if ( this.props.fetchingSiteData ) {
-					return '';
-				}
-
-				return (
-					<Button
-						compact={ true }
-						primary={ true }
-						href={ 'https://jetpack.com/redirect/?source=upgrade-seo&site=' + this.props.siteRawUrl + '&feature=advanced-seo' }
-					>
-						{ __( 'Upgrade', { context: 'Caption for a button to purchase a paid feature.' } ) }
-					</Button>
-				);
-			}
-
-			if ( 'wordads' === feature ) {
-				if ( this.props.fetchingSiteData ) {
+				if ( hasFree ) {
+					if ( this.props.isCompact ) {
+						return this.getProActions( 'free' );
+					}
 					return '';
 				}
 
-				return (
-					<Button
-						compact={ true }
-						primary={ true }
-						href={ 'https://jetpack.com/redirect/?source=upgrade-ads&site=' + this.props.siteRawUrl + '&feature=jetpack-ads' }
-					>
-						{ __( 'Upgrade', { context: 'Caption for a button to purchase a paid feature.' } ) }
-					</Button>
-				);
-			}
-
-			if ( 'google-analytics' === feature && ! includes( [ 'jetpack_business', 'jetpack_business_monthly' ], sitePlan.product_slug ) ) {
-				if ( this.props.fetchingSiteData ) {
-					return '';
+				const akismetData = this.props.getAkismetData();
+				if ( 'invalid_key' === akismetData ) {
```

Comment:
This shold use `isAkismetKeyValid()` since it returns only the data necessary for key validation. `getAkismetData()` returns all the data for prevented spam and it's a larger amount of data.